app/models/search_result.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class SearchResult < ApplicationRecord
+  enum search_engines: { google: 0, bing: 1 }
+
+  validates :keyword, presence: true
+  validates :search_engine, presence: true
```

Comment:
We might want to validate that `search_engine` is one of the valid one. Even though `UploadValidator` has some validation for it, for data integrity, the model must have such validation, i.e., a developer could insert new data in that table without passing through the `Upload` mode.